tools/driver/p4c_src/main.py

```diff
@@ -224,39 +307,152 @@ def main():
         if (pattern.match(opts.target + '-' + opts.arch)):
             backend = target
             break
-    if backend == None:
+    if backend is None:
         parser.error("Unknown backend: {}-{}".format(str(opts.target),
                                                      str(opts.arch)))
+    error_count = 0
+
+    JSON_input_specified = env_indicates_developer_build and opts.json_source
+### ^^^^^^^^^^^^^^^^^^^^: DRY
```

Comment:
I have no idea what DRY stands for

Reply:
Quoting `https://en.wikipedia.org/wiki/Don%27t_repeat_yourself`:

     "Don't repeat yourself" (DRY) is a [principle](https://en.wikipedia.org/wiki/Principle#Principle_as_axiom_or_logical_fundament) of [software development](https://en.wikipedia.org/wiki/Software_development_process) aimed at reducing repetition of software patterns,[[1]](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself#cite_note-1) replacing it with abstractions or using [data normalization](https://en.wikipedia.org/wiki/Data_normalization) to avoid redundancy.


Do you think I should spell it out in the [driver] comment[s]?

Reply:
I added expansions/explanations of “DRY” in the 2 places it appears [both in comments] in “main.py” [in my WIP Git], and left alone “DRYified”.  I guess that`s good enough, so I`m going to tell GitHub to “resolve” this one.